.changeset/spicy-cameras-sleep.md

```diff
@@ -0,0 +1,16 @@
+---
+"@logto/console": minor
+"@logto/core": minor
+---
+
+add access deny method to the custom token claims script
+
+Introduce a new `api` parameter to the custom token claims script. This parameter is used to provide more access control context over the token exchange process.
+Use `api.denyAccess()` to reject the token exchange request. Use this method to implement your own access control logics.
+
+```javascript
+const getCustomJwtClaims: async ({api}) => {
+  // Reject the token exchange request
+  api.denyAccess('Access denied');
```

Comment:
the comment better to say the message is customizable

Reply:
nit:
```suggestion
  return api.denyAccess('Access denied');
```

Reply:
the return type of `getCustomJwtClaims` must be a JSON object.